File: Part 2 - Regression/Section 9 - Random Forest Regression/Random_forest.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jul 16 12:12:46 2018

@author: Vinayak_S
"""

#Importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#Reading data 
df = pd.read_csv('Position_Salaries.csv')
X = df.iloc[:,1:2].values
y = df.iloc[:,-1].values

#Implementing Random Forest Method
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators = 300, random_state = 0)
regressor.fit(X,y)

#Predicting using Random Forest Regression
y_pred = regressor.predict(6.5)

#Visualising the result
X_grid = np.arange(min(X),max(X),0.01).reshape(-1,1)
plt.scatter(X,y)
plt.plot(X_grid, regressor.predict(X_grid))
plt.show()

File: Part 2 - Regression/Section 9 - Random Forest Regression/Random_forest.R
#Reading the data
df = read.csv('Position_Salaries.csv')
df = df[2:3]

#Fitting Random Forest Regression
# install.packages('randomForest')
set.seed(1234)
library(randomForest)
regressor = randomForest(x = df[1],
                         y = df$Salary,
                         ntree = 100)

#Predicting the value using Regression
y_pred = predict(regressor, newdata = data.frame(Level = 6.5))

#Visualisation of the Random Forest Regression
library(ggplot2)
X_grid = seq(min(df$Level),max(df$Level),0.01)
X_grid = data.frame(Level =X_grid)
ggplot()+
  geom_point(aes(x = df$Level, y = df$Salary),
             colour = 'red')+
  geom_line(aes(x = X_grid$Level, y = predict(regressor, newdata = X_grid)),
            colour = 'blue')+
  ggtitle('Level vs Salary (Random Forest Regression Model')+
  xlab('Level')+
  ylab('Salary')
